### PR TITLE
Wrapper classes for UI Lists etc that ensure updates on JavaFX thread

### PR DESCRIPTION
**Pull request by r0light**
Fixes [#5108](https://github.com/JabRef/jabref/issues/5108) 
Fixes [#5109 ](https://github.com/JabRef/jabref/issues/5109)

Furthermore, the threading issue part of #4958 is fixed by using `UiThreadStringProperty`

New classes are added to `gui.util.uithreadaware` which enable a wrapper class around `ObservableList` which ensures that changes are propagated on the JavaFX Application thread.

I started with `UiThreadListDecorator`, but others like `UiThreadMapDecorator` could be implemented similarly.

----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [x] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: src/main/java/org/jabref/gui/util/uithreadaware/UiThreadHelper.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package org.jabref.gui.util.uithreadaware;
+
+import javafx.application.Platform;
+
+public class UiThreadHelper {
```
[the remark this answers is not in the document]
Comment (r0light) [reply src/main/java/org/jabref/gui/util/uithreadaware/UiThreadHelper.java]: done in 96b9aa5

File: src/main/java/org/jabref/gui/fieldeditors/LinkedFilesEditorViewModel.java [hunk 2]
```diff
@@ -43,7 +44,7 @@
 
 public class LinkedFilesEditorViewModel extends AbstractEditorViewModel {
 
-    private final ListProperty<LinkedFileViewModel> files = new SimpleListProperty<>(FXCollections.observableArrayList(LinkedFileViewModel::getObservables));
+    private final ListProperty<LinkedFileViewModel> files = new SimpleListProperty<>(new UiThreadListDecorator<>(FXCollections.observableArrayList(LinkedFileViewModel::getObservables)));
```
[the remark this answers is not in the document]
Comment (r0light) [reply src/main/java/org/jabref/gui/fieldeditors/LinkedFilesEditorViewModel.java]: ok, done in 96b9aa5

**Comment by r0light** [comment]
Ok, it took me while to reproduce #4958 reliably but now i added a `UiThreadStringProperty` in `PersonsEditor` which wraps the original property and the issue should be fixed now.

@tobiasdiez should we use as few as possible new UI thread aware classes only where necessary or should i add it consistently to classes in `gui.fieldeditors` ?

File: src/main/java/org/jabref/gui/fieldeditors/PersonsEditor.java [hunk 3]
```diff
@@ -28,7 +30,8 @@ public PersonsEditor(final Field field,
                 ? new EditorTextField()
                 : new EditorTextArea();
 
-        textInput.textProperty().bindBidirectional(viewModel.textProperty());
+        decoratedStringProperty = new UiThreadStringProperty(viewModel.textProperty());
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/fieldeditors/PersonsEditor.java]: Is it necessary to keep a track of the reference? If not I would prefer if `decoratedStringProperty` is converted to a local variable (or be inlined completely).

**Comment by tobiasdiez** [comment]
Thanks! I think it is fine if we add the wrappers only when they are really necessary. So this PR is ready from your side?

File: src/main/java/org/jabref/gui/fieldeditors/PersonsEditor.java (hunk 3 above)
Comment (r0light) [reply src/main/java/org/jabref/gui/fieldeditors/PersonsEditor.java]: In fact, i tried it with an inlined approach first, but the list of files did not update itself then.
I did not investigate it further, but i think it has to do with `WeakReference`s used inside `Bindings.bindContentBidirectional`.

**Comment by r0light** [comment]
> 
> 
> Thanks! I think it is fine if we add the wrappers only when they are really necessary. So this PR is ready from your side?

Alright, not quite, i want have another look at it and also add some comments to the new classes.

**Comment by r0light** [comment]
Ok, from my side this would be ready to merge.
@tobiasdiez thanks for the help and feedback for this!

**Comment by Siedlerchr** [comment]
I guess #5010 is obsolete now?

**Pull request merged**